metpy/calc/indices.py

```diff
@@ -51,3 +52,64 @@ def precipitable_water(dewpt, p, top=400 * units('hPa')):
     pw = -1. * (np.trapz(w.magnitude, pres_layer.magnitude) * (w.units * pres_layer.units) /
                 (g * rho_l))
     return pw.to('millimeters')
+
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[pressure]', '[length]', '[length]', '[length]')
+def mean_wind_pressure_weighted(u, v, p, hgt, top, bottom=None, interp=False):
+    r"""Calculate pressure-weighted mean wind through a layer.
+
+    Layer top and bottom specified in meters AGL. Options are included to use only
+    observed winds or interpolate through the layer to match NSHARP/SharpPy's output.
+
+    Parameters
+    ----------
+    u : array-like
+        U-component of wind.
+    v : array-like
+        V-component of wind.
+    p : array-like
+        Atmospheric pressure profile
+    hgt : array-like
+        Heights from sounding
+    top: `pint.Quantity`
+        The top of the layer in meters AGL
+    bottom: `pint.Quantity`, optional
+        The bottom of the layer in meters AGL.
+        Default is the surface.
+    interp: boolean, optional
```

Comment:
Sounds fine to me. I think `interpolate` means exactly what it does in `get_layer`... I'm not even sure that I think their 1hPa interpolation method from Sharp is valid from a math standpoint. Anyhow, should be note in notes that it may be slightly different from Sharp because of integration methods?